### PR TITLE
Generate Storybook URLs using v5+ structure

### PR DESCRIPTION
**Pull request by yuqu**
Fixes https://github.com/zeplin/cli-connect-storybook-plugin/issues/11

### CODE REVIEW

**Comment by jgarplind** [comment]
LGTM.

Intent is aligned with need.

Solution confirmed to work by running `npm link` in own repo.

File: src/index.ts [hunk 1]
```diff
@@ -86,18 +88,20 @@ export default class implements ConnectPlugin {
             });
 
             if (matchedStory) {
-                links.push(this.createLink(matchedStory.kind, matchedStory.name));
+                const { storyId, kind, name } = matchedStory;
+                links.push(this.createLink({ storyId, kind, name }));
```
Comment (skarahoda) [inline src/index.ts]:
```suggestion
                links.push(this.createLink(matchedStory));
```

File: src/index.ts [hunk 2]
```diff
@@ -86,18 +88,20 @@ export default class implements ConnectPlugin {
...
             }
         }
 
         const sbManualConf = componentConfig.storybook;
```
Comment (skarahoda) [inline src/index.ts]:
```suggestion
        const { kind, stories } = componentConfig.storybook || {};
```

File: src/index.ts [hunk 3]
```diff
@@ -111,17 +115,25 @@ export default class implements ConnectPlugin {
         return this.stories.length > 0;
     }
 
-    private createLink(kind: string, name?: string): Link {
-        const urlEncodedKind = encodeURIComponent(kind);
+    private createLink(params: { storyId?: string; kind: string; name?: string }): Link {
```
Comment (skarahoda) [inline src/index.ts]:
```suggestion
    private createLink({ storyId, kind, name }: { storyId?: string; kind: string; name?: string }): Link {
```

File: src/index.ts (hunk 2 above)
Comment (yuqu) [reply src/index.ts]: I cannot destruct it since `componentConfig.storybook` is `StorybookComponentConfig | undefined`

File: src/index.ts [hunk 4]
```diff
@@ -111,17 +115,25 @@ export default class implements ConnectPlugin {
...
-            preparedUrl = urljoin(this.targetUrl, `?selectedKind=${urlEncodedKind}`);
+            const encodedKind = encodeURIComponent(kind);
+
+            if (name) {
+                url = join(
+                    this.targetUrl,
+                    `/?selectedKind=${encodedKind}&selectedStory=${encodeURIComponent(name)}`
+                );
+            } else {
+                url = join(this.targetUrl, `/?selectedKind=${encodedKind}`);
```
Comment (skarahoda) [inline src/index.ts]: we can use `query` parameter like: `urlJoin(this.targetUrl, { query: { selectedKind: encodedKind } });`

File: src/index.ts [hunk 5]
```diff
@@ -111,17 +115,25 @@ export default class implements ConnectPlugin {
...
+        const { storyId, kind, name } = params;
 
-        let preparedUrl;
-        if (name) {
-            const urlEncodedName = encodeURIComponent(name);
-            preparedUrl = urljoin(this.targetUrl, `?selectedKind=${urlEncodedKind}&selectedStory=${urlEncodedName}`);
+        let url: string;
+        if (storyId) {
+            url = join(this.targetUrl, `/?path=/story/${storyId}`);
         } else {
```
Comment (skarahoda) [inline src/index.ts]: We can change like: 
```
        } else if (name) {
.....
```

File: src/index.ts (hunk 2 above)
Comment (skarahoda) [reply src/index.ts]: I think you can destruct if you add `|| {}`. WDYT?

File: src/index.ts (hunk 4 above)
Comment (yuqu) [reply src/index.ts]: I don't think this is necessary. 

Firstly, it just makes me to repeat put same option inputs on every call (proper-url-join has troublesome API definition, the problem is not type definition but the implementation is not suitable for rest params input definition).

Secondly, it is more visible to maintenance. No need to delegate SB URL specifics into a 3rd party lib. I just use it to ensure the final concatenated string is a good lookin valid URL without redundant characters(e.g. excessive slashes).

Comment (skarahoda) [reply src/index.ts]: 👌

**Pull request merged**